### PR TITLE
Fixing the docker build action

### PR DESCRIPTION
**Pull request by shimwell**
## Description
Tiny change to the docker building ci that:
-updates to use the latest tag-push-action from v1 to v2
-we had two steps with the same name so I've renamed the second one so that it better describes what is happening.
-added some comments to the steps that helps with issue #1425 

## Changes
Updated docker build CI

## Behavior
failing CI
https://github.com/pyne/pyne/actions/workflows/docker_publish.yml

## Other Information
Currently a blocking issue for the next DAGMC release

## Changelog file
All pull requests are required to update the [CHANGELOG](https://github.com/pyne/pyne/blob/develop/CHANGELOG.rst) file with the PR.  Your update can take different forms including:

* creating a new entry describing your change, including a reference to this pull request number
* adding a reference to your pull request number to an exist entry if that entry can include your changes

### CODE REVIEW

**Comment by gonuke** [comment]
Thanks @Shimwell - can you add a note in the CHANGELOG please?

**Comment by gonuke** [comment]
This workflow only runs on push, so it won't run on the PR.

I looked to your repo/branch, but it failed because image repos can't have an upper case name...

**Comment by shimwell** [comment]
> This workflow only runs on push, so it won't run on the PR.
> 
> I looked to your repo/branch, but it failed because image repos can't have an upper case name...

I've changed my username to be all lowercase now and have retriggered the action on my fork. 

https://github.com/shimwell/pyne/actions/runs/1397725708

It looks like everything passed ✔️ but the last stage didn't run because of the ```if github.repository_owner == 'pyne' ``` part

**Comment by gonuke** [comment]
I think we'll just need to try it.

**Review by gonuke — CHANGES_REQUESTED** [review]
It looks like our standard BuildTest runs in 5 varieties, including one without pymoab or DAGMC.  However, this workflow no longer builds a pyne_Dept image without pymoab and DAGMC. There is one in the packages, so that old one will be used each time.  It may not change much, but we should probably rebuild it as well as part of one of these steps.

**Comment by bam241** [comment]
> > This workflow only runs on push, so it won't run on the PR.
> > I looked to your repo/branch, but it failed because image repos can't have an upper case name...
> 
> I've changed my username to be all lowercase now and have retriggered the action on my fork.
> 
> https://github.com/shimwell/pyne/actions/runs/1397725708
> 
> It looks like everything passed ✔️ but the last stage didn't run because of the `if github.repository_owner == 'pyne' ` part

You could change temporary it to `shimwell`

File: .github/workflows/docker_publish.yml
Comment (bam241) [inline .github/workflows/docker_publish.yml]: I think I disagree with this.
I believe we did build 5 Img and tried to push 6.
the one this blank option would allow to build:
 - `no-pymoab -- no-dagmc`
 -  `pymoab -- no-dagmc`
which we should be building...
but the option `no-pymoab` & `dagmc` does not exist and should ignore in the matrix....

Comment (bam241) [reply .github/workflows/docker_publish.yml]: see `ctr: failed to resolve reference "ghcr.io/pyne/ubuntu_18.04_py3_dagmc_pyne-deps:ci_testing`from https://github.com/pyne/pyne/runs/3690294186?check_suite_focus=true

**Comment by bam241** [comment]
If I recall correctly:
the first job builds the following option:
- py3
- py3+pymoab+apt-hdf5

the second one:
- py3+pymoab+dagmc(+apt-hdf5)
- py3+pymoab+dagmc+hdf5-1.12
- py3+pymoab+dagmc(+apt-hdf5)+openmc


I think the failing push is on py3+dagmc(+apt-hdf5)


(Note that apt-hdf5 is not tagged on the docker img as it is the default. We added the hdf5-1.12 to test the build that was problematic (it need special building flags...)
I hope this helps

File: .github/workflows/docker_publish.yml
Comment (shimwell) [reply .github/workflows/docker_publish.yml]: > > > This workflow only runs on push, so it won't run on the PR.
> > > I looked to your repo/branch, but it failed because image repos can't have an upper case name...
> > 
> > 
> > I've changed my username to be all lowercase now and have retriggered the action on my fork.
> > https://github.com/shimwell/pyne/actions/runs/1397725708
> > It looks like everything passed ✔️ but the last stage didn't run because of the `if github.repository_owner == 'pyne' ` part
> 
> You could change temporary it to `shimwell`

Thanks, Bam, I've changed to shimwell repo owner so I can test the whole workflow. I will change it back once everything is ready

**Comment by shimwell** [comment]
Super thanks for all the tips, I've been tinkering and am looking forward to seeing (after the ci finishes) to see if I've got all the right changes implemented.

**Comment by shimwell** [comment]
ok i think this is working as required. The latest action build and it did the last step as well.

https://github.com/shimwell/pyne/actions/runs/1401917906

before merging if we are happy with this then I just need to change the name of the repo back to ```pyne```

File: .github/workflows/docker_publish.yml
Comment (shimwell) [inline .github/workflows/docker_publish.yml]: This just needs to be changed back, this was a temporary change made so that I could test the entire ci action

```suggestion
    if: ${{ github.repository_owner == 'pyne' }}
```

**Review by gonuke — CHANGES_REQUESTED** [review]
This appears to work, but for maintainability, I would like the matrix/include for `BuildTest` to match the matrix/include for the last action.

Right now they produce the same combinations, but those combinations are generated with different syntax, which will sew confusion in the future

**Comment by shimwell** [comment]
Matrix for the last two stages are now the same and the actions work https://github.com/shimwell/pyne/actions/runs/1404918978

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
I'm ready to merge if you want to make that last quick change back to the PyNE repo

**Pull request merged**
